openshift/buildconfig-template.yaml

```diff
@@ -0,0 +1,75 @@
+---
+apiVersion: v1
+kind: Template
+metadata:
+  annotations:
+    description: "This is Nepthys BuildConfig"
+    openshift.io/display-name: "Thoth: Nepthys BuildConfig"
+    tags: "thoth,ai-stacks,nepthys"
+    template.openshift.io/documentation-url: "https://github.com/Thoth-Station/"
+    template.openshift.io/long-description: "This template defines resources needed to deploy Thoth Nepthys to OpenShift.\n"
+    template.openshift.io/provider-display-name: "Red Hat, Inc."
+    version: "1.0.0"
+  labels:
+    app: thoth
+    component: nepthys
+    template: nepthys-buildconfig
+  name: nepthys-buildconfig
+objects:
+  - apiVersion: build.openshift.io/v1
+    kind: BuildConfig
+    metadata:
+      labels:
+        app: nepthys
+      name: nepthys-job
+    spec:
+      output:
+        to:
+          kind: ImageStreamTag
+          name: "nepthys-job:${IMAGE_STREAM_TAG}"
+      resources:
+        limits:
+          cpu: "1"
+          memory: 1Gi
+        requests:
+          cpu: "1"
+          memory: 1Gi
+      runPolicy: Serial
+      source:
+        git:
+          ref: "${GITHUB_REF}"
+          uri: "${GITHUB_URL}"
+        type: Git
+      strategy:
+        dockerStrategy:
+          dockerfilePath: Dockerfile
+          noCache: true
+        env:
+          - name: ENABLE_PIPENV
+            value: "1"
+          - name: UPGRADE_PIP_TO_LATEST
+            value: ""
+        type: Docker
+      triggers:
+        - imageChange: {}
+          type: ImageChange
+        - generic:
+            allowEnv: false
+            secret: secret101
```

Comment:
Not to have secrets directly in templates, you can create parameters for them and set them from ansible playbooks (you can use this as a default value). We will need to adjust this on "prod" provisioning anyway, to have it properly "secret".